src/Microsoft.AspNetCore.Mvc.Core/Authorization/AuthorizeFilter.cs

```diff
@@ -75,7 +75,6 @@ public AuthorizeFilter(AuthorizationPolicy policy)
 
             // Note: Default Anonymous User is new ClaimsPrincipal(new ClaimsIdentity())
             if (httpContext.User == null ||
```

Comment:
Hehe yeah, people do unexpected things sometimes :smile: 

FWIW, I'd personally remove this `null` check and treat null principals like empty ones, specially since the authorization service is fine with null values.